common/txmgr/types/tx.go

```diff
@@ -66,6 +66,10 @@ func (s TxAttemptState) String() (str string) {
 }
 
 type TxRequest[ADDR types.Hashable, TX_HASH types.Hashable] struct {
+	// Unique ID set by external components to prevent double send
+	// If present in TxRequest, TXM will search for Tx with ID
+	// If found, it will return the existing Tx, otherwise it will create a new Tx
+	RequestID        *string
```

Comment:
Can we try to think of a name for this field that doesn't result in stuttering? `TxRequest.RequestID`

Reply:
> ...TXM will search for Tx with ID
> // If found, it will return the existing Tx

What does 'return' mean in this context? Can we describe the effect here without including any implementation details? What if rest of the fields are different? Is that info discarded, or is the existing one updated?

Reply:
Don't have strong opinion on the name. RequestId was already used in the TxMeta field by VRF, so made sense to reuse the same name. Is stuttering a real problem? I don't see people having to read out the whole thing.
I could think of other alternatives:
- UniqueId
- IdempotencyKey

For the comment on this field, how about changing to:
// RequestId is a globally unique ID set by the caller, to prevent accidental creation of duplicated Txs during retries or crash recovery.
// If this field is set, the TXM will first search existing Txs with this field.
// If found, it will return the existing Tx, without creating a new one. TXM will not validate or ensure that existing Tx is same as the incoming TxRequest.
// If not found, TXM will create a new Tx.
// If requestID is set to null, TXM will always create a new Tx.
// Callers must ensure that they use a globally unique id while setting this field.

Reply:
> Is stuttering a real problem?

Eliminating redundant words is one of the few pieces of advice on naming: https://go.dev/doc/effective_go#names

In this case, I think it reveals that there is more to say due to the overloaded terminology obscuring things. A different word like `Idempotency` definitely adds value here :+1: 

> // Callers must ensure that they use a globally unique id while setting this field.

Do we need to protect them from messing this up? e.g. by using a `uuid.UUID`

Reply:
I'll go ahead and change it to `IdempotencyKey` then and add the new comment above. I wouldn't be opposed to changing it to `uuid` either. It'd ensure unique values are being used across services that need this.